manim/scene/scene_file_writer.py

```diff
@@ -482,11 +484,14 @@ def is_already_cached(self, hash_invocation):
         :class:`bool`
             Whether the file exists.
         """
-        path = os.path.join(
-            self.partial_movie_directory,
-            "{}{}".format(hash_invocation, self.movie_file_extension),
-        )
-        return os.path.exists(path)
+        if hasattr(self, "partial_movie_directory"):
+            path = os.path.join(
+                self.partial_movie_directory,
+                "{}{}".format(hash_invocation, self.movie_file_extension),
+            )
+            return os.path.exists(path)
+        else:
+            return False  # If there is no partial movie directory, then the file is not cached
```

Comment:
This part allows to use «save_last_frame» without «write_to_movie». If you use that, the partial_movie_directory variable is never set, but the program checks if the movie has been generated and fails beccause there is no partial_movie_directory. So if there is no partial_movie_directory, the obvious choice is that the file has not been generated